### PR TITLE
Feat/add command to start service registry contracts locally

### PR DESCRIPTION
**Pull request by 0xArdi**
## Proposed changes

Adds support for running a network with the service registry contract locally, replacing the staging chain.

Related to #1241.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [x] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: autonomy/cli/develop.py [hunk 1]
```diff
@@ -20,10 +20,41 @@
...
+    required=False,
+    default=DEFAULT_SERVICE_REGISTRY_CONTRACTS_IMAGE,
+)
+def run_service_locally(image: str) -> None:
+    """Run the service registry contracts on a local network."""
+    client = from_env()
+    container = client.containers.run(
+        image=image,
+        detach=True,
+        network_mode="host",
```
Comment (0xArdi) [inline autonomy/cli/develop.py:47]: no need to configure port binding

File: autonomy/cli/develop.py [hunk 2]
```diff
@@ -20,10 +20,41 @@
...
+    )
+    try:
+        for line in client.api.logs(container.id, follow=True, stream=True):
+            click.echo(line.decode())
```
Comment (DavidMinarsch) [inline autonomy/cli/develop.py:51]: Double check with @angrybayblade , there were some cross-system problems once with similar code

**Review by DavidMinarsch — COMMENTED** [review]
Please add a test

**Review by DavidMinarsch — DISMISSED** [review]
LGTM

**Review by DavidMinarsch — APPROVED** [review]

**Review by angrybayblade — APPROVED** [review]
LGTM

**Pull request merged**